lib/rules/alpha-value-notation/index.js

```diff
@@ -43,11 +42,23 @@ function rule(primary, options, context) {
 
 		if (!validOptions) return;
 
+		const optionFuncs = Object.freeze({
+			number: {
+				expFunc: isNumber,
+				fixFunc: asNumber,
+			},
+			percentage: {
+				expFunc: isPercentage,
+				fixFunc: asPercentage,
+			},
+		});
+
```

Comment:
[note] This `optionFuncs` move aims to reduce needless object creation within the `walkDecls` callback.